pkg/hubble/container/ring.go

```diff
@@ -31,6 +32,67 @@ var (
 	ErrInvalidRead = errors.New("read position is no longer valid")
 )
 
+// capacity implements Capacity.
+type capacity uint16
```

Comment:
Is there a particular reason to use `uint16`? Otherwise I'd use an `int` as it avoids lots of typecasts. Note that Go doesn't use unsigned integers for lengths, e.g. the builtin `len` function returns an `int`.

Reply:
> it avoids lots of typecasts

What do you mean? There's only one typecast that could be avoided by using an int. I know that Go doesn't use unsigned integers for lengths (and capacities) but the ring buffer's implementation of `Cap()` and `Len()` returns an `uint64` in any case (admittedly, this could be fixed but would require a type cast in the ring implementation).

Reply:
OK, "lots" of typecasts was an overstatement :) I generally feel that unsigned and sized integer types should only be used when absolutely necessary however, but this is merely a personal preference.

Reply:
I agree with you on this and use `int` for sizes/capacities as this is what is done throughout Go's codebase but in this particular case, I think using `uint16` makes sense. See also [my comment below](https://github.com/cilium/cilium/pull/13894#discussion_r519681527).